### DISCUSSION TITLE
execute.nothrow - a bit which does not work ?

### DISCUSSION THREAD

**Discussion by lejeczek** [rakshasa/rtorrent#1229] [General]
Hi guys.

I have these two lines in my .rc:
...
execute.nothrow = chmod, 777, "/var/run/rtorrent/rpc.socket"
execute.nothrow = chgrp, nginx, "/var/run/rtorrent/rpc.socket"

and result of first is as expected but second one, seems to do nothing, nor 'rtorrent' logs any errors.

version 0.9.8

All comments are much welcomed.
many thanks, L.

**Comment by kannibalox** [comment]
`execute.nothrow` simply stores the return code of the command, and won't do anything with the result. You can try using `execute.throw`, which will throw an RPC error if the command fails, but depending on your end goal you may need outsource the error handling to a bash script.

**Comment by lejeczek** [comment]
I resolved this and would be glad for a comment from author/devel.

'rtorrent' when run as daemon via/by 'systemd' turns out - because of 'systemd' and restriction, confinements & limitation it manages processes with - service unit needs:

[Unit]
Description=rTorrent System Daemon
After=network.target

[Service]
Type=simple
User=rtorrent
Group=nginx <= !! HERE as opposed to user's primary group, if you want your Nginx to be able to talk to the socket that is

ExecStartPre=-/bin/rm -f /home.sysop/rtorrent/.session/rtorrent.lock
ExecStart=/usr/bin/rtorrent -n -o import=/home.sysop/rtorrent/bin/conf/rtorrent.rc
Restart=on-failure
RestartSec=3

[Install]
WantedBy=multi-user.target

then 'chgrp' in 'execute.nothrow' does what is expected of it.

many thanks, L.

**Reply by kannibalox** [reply]
> I resolved this and would be glad for a comment from author/devel.

What kind of comment are you looking for? There were multiple ways to solve this problem and it seems you found a satisfactory one. rakshasa isn't very active nowadays, and this doesn't seem to particularly require their expertise, so if you have any questions I or one of the other community members will probably be able get back to you more quickly.

**Reply by lejeczek** [reply]
Comments from devel/authors are always best - well.. most of time are, exceptions are underwhelming developers/programmers - for such authors can - most of the time do - explain(to us users, non-programmers) what is really happening under the hood.
My solution as I described it - I might see & understand what is happening but.. at the higher/est level, not what happens under the hood.
That is what hope for when I hope for author(s) comment but, it's okey & I don't despair if those are not given.
thanks, L.
